### PR TITLE
Fixes #35488 - Fix snapshots:generate task

### PR DESCRIPTION
**Pull request by stejskalleos**
Fixed issues:
* Generating snapshot for provision/coreos_provision.erb
* Generating snapshot for PXEGrub2/kickstart_default_pxegrub2.erb

```
RAILS_ENV=test bundle exec rails snapshots:generate
```

### CODE REVIEW

**Comment by stejskalleos** [comment]
[test integration]
[test katello]
[test unit]

**Comment by Ron-Lavi** [comment]
ok to test

**Comment by stejskalleos** [comment]
[test unit]

**Comment by Ron-Lavi** [comment]
[test unit]

**Review by Ron-Lavi — APPROVED** [review]
Thanks for fixing it @stejskalleos !!

File: lib/tasks/snapshots.rake [hunk 1]
```diff
@@ -16,7 +16,15 @@ namespace :snapshots do
         nil
       end
     end
+
+    module BaseMacrosStub
+      def dns_lookup(_name_or_ip)
+        'foreman.example.com'
```
Comment (ekohl) [inline lib/tasks/snapshots.rake:22]: This is a poor implementation. dns_lookup for a name should return an IP.

**Pull request merged**